aiosmtpd/smtp.py

```diff
@@ -591,6 +593,32 @@ def _syntax_available(self, method):
         status = await self._call_handler_hook('RSET')
         await self.push('250 OK' if status is MISSING else status)
 
+    def _parse_xclient_attributes(self, attributes):
+        """
+        Parse key-value arguments and return OrderedDict on success.
+        Raise ValueError on wrong arguments.
+        """
+        result = collections.OrderedDict()
```

Comment:
Given that in Python 3.7, built-in dicts are guaranteed to be ordered, and in 3.6 they are *effectively* ordered, should we use `OrderedDict` in those cases?  You'd probably have to check Python's major/minor version to know which one to use, so I wouldn't be opposed to an argument of using `OrderedDict` for consistency.  It's just that when we inevitably drop <=3.5 support, we won't need it.